pkg/gitprovider/types.go

```diff
@@ -17,16 +17,24 @@ type GitUser struct {
 	Email    string `json:"email" validate:"required"`
 } // @name GitUser
 
+type CloneTarget string // @name CloneTarget
+
+const (
+	CloneTargetBranch CloneTarget = "branch"
+	CloneTargetCommit CloneTarget = "commit"
+)
+
 type GitRepository struct {
-	Id       string  `json:"id" validate:"required"`
-	Url      string  `json:"url" validate:"required"`
-	Name     string  `json:"name" validate:"required"`
-	Branch   *string `json:"branch,omitempty" validate:"optional"`
-	Sha      string  `json:"sha" validate:"required"`
-	Owner    string  `json:"owner" validate:"required"`
-	PrNumber *uint32 `json:"prNumber,omitempty" validate:"optional"`
-	Source   string  `json:"source" validate:"required"`
-	Path     *string `json:"path,omitempty" validate:"optional"`
+	Id       string      `json:"id" validate:"required"`
+	Url      string      `json:"url" validate:"required"`
+	Name     string      `json:"name" validate:"required"`
+	Branch   *string     `json:"branch,omitempty" validate:"optional"`
+	Sha      string      `json:"sha" validate:"required"`
+	Owner    string      `json:"owner" validate:"required"`
+	PrNumber *uint32     `json:"prNumber,omitempty" validate:"optional"`
+	Source   string      `json:"source" validate:"required"`
+	Path     *string     `json:"path,omitempty" validate:"optional"`
+	Target   CloneTarget `json:"clonetarget,omitempty" validate:"optional"`
```

Comment:
Change the json prop from `clonetarget` to `cloneTarget` here and in the db type